### PR TITLE
DT-916 Bug where response from update was a list not single item

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/MessageIntegrationTest.kt [hunk 1]
```diff
@@ -17,7 +17,6 @@ import javax.inject.Inject
   "prisontoprobation.hold-back.duration=0m"
 ])
 @DirtiesContext
-@ActiveProfiles(profiles = ["test-int"])
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/MessageIntegrationTest.kt:20]: got rid of this nonsense fix at some time

File: src/test/resources/application-test.yml [hunk 2]
```diff
@@ -1,3 +1,7 @@
+spring:
+  profiles:
+    include: stdout
```
Comment (andymarke) [inline src/test/resources/application-test.yml:3]: nice to see stdout when running tests

File: src/test/resources/mappings/putOffenderNomsNumber_A9999DY.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (andymarke) [inline src/test/resources/mappings/putOffenderNomsNumber_A9999DY.json:1]: and this was missing completely ..

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/services/CommunityService.kt [hunk 4]
```diff
@@ -72,12 +72,12 @@ class CommunityService(@Qualifier("probationApiWebClient") private val webClient
...
   }
 
-  fun replaceProbationOffenderNo(oldOffenderNo: String, newOffenderNo: String) : IDs?{
+  fun replaceProbationOffenderNo(oldOffenderNo: String, newOffenderNo: String) : List<IDs>?{
     return webClient.put()
         .uri("/secure/offenders/nomsNumber/$oldOffenderNo/nomsNumber")
         .bodyValue(UpdateOffenderNomsNumber(nomsNumber = newOffenderNo))
         .retrieve()
-        .bodyToMono(IDs::class.java)
+        .bodyToMono(object : ParameterizedTypeReference<List<IDs>>() {})
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/services/CommunityService.kt:80]: Does the class you deserialize to represent one or more Ids?  Seems a bit weird possibly returning a list of plural items

Comment (andrewrlee) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/services/CommunityService.kt:80]: oh guess its `IDs` as in there is more than one type of id for a specific offender there.

Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/services/CommunityService.kt:80]: yeah, a list of a set of identifiers

**Pull request merged**